x-pack/legacy/plugins/upgrade_assistant/server/np_ready/routes/cluster_checkup.test.ts

```diff
@@ -90,35 +92,23 @@ describe('cluster checkup API', () => {
       e.status = 403;
 
       MigrationApis.getUpgradeAssistantStatus.mockRejectedValue(e);
-      const resp = await server.inject({
-        method: 'GET',
-        url: '/api/upgrade_assistant/status',
-      });
+      const resp = await serverShim.router.getHandler({
+        method: 'get',
+        pathPattern: '/api/upgrade_assistant/status',
+      })(ctxMock, createRequestMock(), kibanaResponseFactory);
 
-      expect(resp.statusCode).toEqual(403);
+      expect(resp.status).toEqual(403);
     });
 
     it('returns an 500 error if it throws', async () => {
       MigrationApis.getUpgradeAssistantStatus.mockRejectedValue(new Error(`scary error!`));
-      const resp = await server.inject({
-        method: 'GET',
-        url: '/api/upgrade_assistant/status',
-      });
 
-      expect(resp.statusCode).toEqual(500);
-    });
-
-    it('returns a 426 if EsVersionCheck throws', async () => {
-      (EsVersionPrecheck.method as jest.Mock).mockRejectedValue(
-        new Boom(`blah`, { statusCode: 426 })
-      );
-
-      const resp = await server.inject({
-        method: 'GET',
-        url: '/api/upgrade_assistant/status',
-      });
+      const resp = await serverShim.router.getHandler({
+        method: 'get',
+        pathPattern: '/api/upgrade_assistant/status',
+      })(ctxMock, createRequestMock(), kibanaResponseFactory);
 
-      expect(resp.statusCode).toEqual(426);
```

Comment:
Isn't it worth keeping this, otherwise you're not testing if your `versionCheckHandlerWrapper` was applied, seems like you should be able to keep this test with your "test harness".

Reply:
So I'm specifically opting out of the wrapper using the jest mock above like:
```ts
jest.mock('../lib/es_version_precheck', () => ({
  versionCheckHandlerWrapper: (a: any) => a,
}));
```

The wrapper logic should be covered by `es_version_precheck.test.ts`. But I hear what you're saying it would be nice to test them in concert but I think since the wrapper is so then it didn't merit a full mock out.